modules/couchbase/couchbase.go

```diff
@@ -48,31 +54,57 @@ type CouchbaseContainer struct {
 	config *Config
 }
 
-// StartContainer creates an instance of the Couchbase container type
-func StartContainer(ctx context.Context, opts ...Option) (*CouchbaseContainer, error) {
+// RunContainer creates an instance of the Couchbase container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*CouchbaseContainer, error) {
 	config := &Config{
-		enabledServices: []service{kv, query, search, index},
-		username:        "Administrator",
-		password:        "password",
-		// defaultImage {
-		imageName: "couchbase:6.5.1",
-		// }
+		enabledServices:  make([]Service, 0),
+		username:         "Administrator",
+		password:         "password",
 		indexStorageMode: MemoryOptimized,
 	}
 
-	for _, opt := range opts {
-		opt(config)
-	}
-
 	req := testcontainers.ContainerRequest{
-		Image:        config.imageName,
-		ExposedPorts: exposePorts(config.enabledServices),
+		// defaultImage {
+		Image: "couchbase:6.5.1",
+		// }
+		ExposedPorts: []string{MGMT_PORT + "/tcp", MGMT_SSL_PORT + "/tcp"},
 	}
 
-	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+	genericContainerReq := testcontainers.GenericContainerRequest{
 		ContainerRequest: req,
 		Started:          true,
-	})
+	}
+
+	for _, srv := range initialServices {
+		opts = append(opts, withService(srv))
+	}
+
+	for _, opt := range opts {
+		opt.Customize(&genericContainerReq)
+
+		// transfer options to the config
+
+		if bucketCustomizer, ok := opt.(bucketCustomizer); ok {
```

Comment:
@fbiville this is how I'm transferring the functional option state to the container config